src/util.ts

```diff
@@ -39,77 +39,73 @@ export function nop() {
+ * If the tsconfig.json file has an "extends" field hop down the dependency tree
+ * until it ends
  */
 
-
-export interface ConfigFile {
-  files?: string[];
-  compilerOptions?: {};
-  include?: string[];
-  exclude?: string[];
-}
-
 export async function getTSConfig(
     rootDir: string, customReadFilep?: ReadFileP): Promise<ConfigFile> {
+  customReadFilep = customReadFilep || readFilep;
   // array to hold accessed files, used to check for circular references
-
-
-
-  const tsconfigPath = path.join(rootDir, 'tsconfig.json');
-
+  const readArr = [''];
+  return await getBase('tsconfig.json', customReadFilep, readArr, rootDir);
+}
+/**
+ *
+ * Recursively iterate through the dependency chain until we reach the end of
+ * the dependency chain or encounter a circular reference
+ * @param filePath Filepath of file currently being read
+ * @param customReadFilep the file reading function being used
+ * @param readFiles an array of the previously read files so we can check for
+ * circular references returns a ConfigFile object containing the data from all
+ * the dependencies
+ */
+async function getBase(
+    filePath: string, customReadFilep: ReadFileP, readFiles: string[],
+    currentDir: string): Promise<ConfigFile> {
   customReadFilep = customReadFilep || readFilep;
-  const json = await customReadFilep(tsconfigPath, 'utf8');
-  let contents = JSON.parse(json);
 
-  const readArr = ['tsconfig.json'];
-  if (contents['extends']) {
-    const extension =
-        await getExtension(contents['extends'], customReadFilep, readArr);
-    contents = combineTSConfig(extension, contents);
-  }
-  return Promise.resolve(contents);
-}
+  filePath = path.resolve(currentDir, filePath);
 
-async function getExtension(
-    filePath: string, customReadFilep: ReadFileP,
-    readFiles: string[]): Promise<ConfigFile> {
-  customReadFilep = customReadFilep || readFilep;
-  if (!filePath) {
-    throw new Error('Undefined passed in');
-  }
   if (readFiles.indexOf(filePath) !== -1) {
-    // did I throw the error correctly?
-    throw new Error('Circular Reference Detected');
+    throw new Error(
+        'Circular reference in ' + readFiles[readFiles.indexOf(filePath)]);
   }
   readFiles.push(filePath);
+
   const json = await customReadFilep(filePath, 'utf8');
   let contents = JSON.parse(json);
 
-  if (contents['extends']) {
-    const nextFile =
-        await getExtension(contents['extends'], customReadFilep, readFiles);
+  if (contents.extends) {
+    const nextFile = await getBase(contents.extends, customReadFilep, readFiles,
+                                                   path.dirname(filePath));
     contents = combineTSConfig(nextFile, contents);
   }
-  // console.log(contents);
-  return Promise.resolve(contents);
+
+  return contents;
 }
 
-// the inherited config file overwrites the base config file's "files",
-// "include", and "exclude" fields and combines compiler options
+/**
+ * takes in 2 config files
```

Comment:
Can you write full sentences in JSDocs? Here as well as in `@param` parameter descriptions.